test/python/transpiler/test_naming_transpiled_circuits.py

```diff
@@ -0,0 +1,157 @@
+        """
+        Given a single circuit and an output name in form of a single element
+        list, this test checks whether the transpiled circuit is mapped with
+        that assigned name in the list.
+        If list has more than one element, then test checks whether the
+        Transpile function raises an error.
+        """
+        qr = QuantumRegister(4)
+        cirkie = QuantumCircuit(qr)
+        cirkie.ccx(qr[0], qr[1], qr[2])
+        basis_gates_var = ['u1', 'u2', 'u3', 'cx']
+        trans_cirkie = transpile(cirkie, basis_gates=basis_gates_var,
+                                 output_names=['transpiled-cirkie'])
+        self.assertTrue(trans_cirkie.name, 'transpiled-cirkie')
+        # If List has multiple elements, transpile function must raise error
+        with self.assertRaises(TranspilerError):
+            transpile(cirkie, basis_gates=basis_gates_var,
+                      output_names=["cool-cirkie", "new-cirkie", "dope-cirkie", "awesome-cirkie"])
+
+    def multiple_circuits_name_singleton(self):
+        """
+        Given multiple circuits and a single string as a name, this test checks
+        whether the Transpile function raises an error.
+        """
+        # first circuit
+        qr1 = QuantumRegister(2)
+        cr1 = ClassicalRegister(2)
+        circ1 = QuantumCircuit(qr1, cr1)
+        circ1.h(qr1[0])
+        circ1.cx(qr1[0], qr1[1])
+        # second circuit
+        qr2 = QuantumRegister(2)
+        cr2 = ClassicalRegister(2)
+        circ2 = QuantumCircuit(qr2, cr2)
+        circ2.measure(qr2, cr2)
+        backendie = BasicAer.get_backend('qasm_simulator')
+        # Raise Error if single name given to multiple circuits
+        with self.assertRaises(TranspilerError):
+            transpile([circ1, circ2], backendie, output_names='circ')
+
+    def multiple_circuits_name_list(self):
+        """
+        Given multiple circuits and a list for output names, if
+        len(list)=len(circuits), then test checks whether transpile func assigns
+        each element in list to respective circuit.
+        If lengths are not equal, then test checks whether transpile func raises
+        error.
+        """
+        # first circuit
+        qr1 = QuantumRegister(2)
+        cr1 = ClassicalRegister(2)
+        circ1 = QuantumCircuit(qr1, cr1)
+        circ1.h(qr1[0])
+        circ1.cx(qr1[0], qr1[1])
+        # second circuit
+        qr2 = QuantumRegister(2)
+        cr2 = ClassicalRegister(2)
+        circ2 = QuantumCircuit(qr2, cr2)
+        circ2.measure(qr2, cr2)
+        # third circuit
+        qr3 = QuantumRegister(2)
+        cr3 = ClassicalRegister(2)
+        circ3 = QuantumCircuit(qr3, cr3)
+        circ3.x(qr3[0])
+        circ3.x(qr3[1])
+        # combining multiple circuits
+        circuits = [circ1, circ2, circ3]
+        backendie = BasicAer.get_backend('qasm_simulator')
+        # equal lengths
+        names = ['awesome_circ1', 'awesome_circ2', 'awesome_circ3']
```

Comment:
These names use underscores but when you check there are hypens, is this translation done somewhere?

Reply:
Good catch @maddy-tod , the assertions below are `self.assertTrue`'s but should probably be `self.assertEqual`s.

Reply:
Awesome catch! The full method is not running because the `test_` prefix is missed!

Reply:
I fixed it in dac91196

Reply:
I fixed all the tests. Please, have a look!